pyro/infer/abstract_infer.py

```diff
@@ -285,23 +286,57 @@ class TracePredictive(TracePosterior):
     :param TracePosterior posterior: trace posterior instance holding
         samples from the model's approximate posterior.
     :param int num_samples: number of samples to generate.
+    :param list(str) resample_sites: sites to resample from in posterior predictive distribution.
```

Comment:
The use case is if we have some data-local sampling sites that depend on global parameters, and which we want to resample instead of reusing parameters optimized/inferred for other data points in the training set. (E.g., we do a local Bernoulli trial whose parameters we learn globally, but which should be re-evaluated independently of whether we just had a small training set where the output was all `1` in the posterior).

Reply:
I'm still not quite sure if that should belong to `TracePredictive` and I don't fully understand the use case. Maybe we can support such a use case with `exclude_sites` instead. The `resample_sites` might be very confusing to users. I would imagine someone specifying `resamples_sites = ["_RETURN"]`  and get a prior predictive rather than a posterior predictive distribution! It will also be nice if we have a test or an example (not necessarily in this PR), that illustrates the use case.

Reply:
It could certainly also be that I am misunderstanding something on how Pyro work, but I will try to add an example test case to clarify my use case 😄.

Reply:
It seems that I could not make up an example that worked like I wanted, so I removed it. Sorry, for the confusion.

Reply:
Np, if you have a use case that is not covered, I would suggest opening an issue so that we can keep track of it as we look to refactor this class.

Reply:
Cool, thanks!